### PR TITLE
fix(core/inlines): render missing inline expansion + better warn msg

### PR DESCRIPTION
**Pull request by marcoscaceres**
When [[[expansion]]] was not found, nothing was being rendered in its place... which is not great. 

Better to render something, even if it's not found.

### CODE REVIEW

**Comment by sidvishnoi** [comment]
I think we should add a `data-original-text` for all the inlines that we parse. We can then use original text in error messages as well as in this case.

**Comment by marcoscaceres** [comment]
Hmm... I went down that route for a bit, but then felt like a lot of duplication. But I guess we always end up back at the original input. 

I also just wanted to fix this one, I think the others may be ok... this is the most "offensive" one.

File: src/core/render-biblio.js [hunk 1]
```diff
@@ -285,15 +285,17 @@ function decorateInlineReference(refs, aliases) {
...
-    const badrefs = [
-      ...document.querySelectorAll(
-        `a.bibref[href="#bib-${ref.toLowerCase()}"]`
-      ),
-    ].filter(({ textContent: t }) => t.toLowerCase() === ref.toLowerCase());
-    const msg = `Bad reference: [\`${ref}\`] (appears ${badrefs.length} times)`;
-    showError(msg, name);
-    console.warn("Bad references: ", badrefs);
+function warnBadRefs(refs) {
+  refs.forEach(({ ref }) => {
```
Comment (sidvishnoi) [inline src/core/render-biblio.js:289]: Nit: While we're here, let's use `for .. of`

**Review by sidvishnoi — APPROVED** [review]

**Comment by sidvishnoi** [comment]
Should use `data-original-text` sometime though. I think it'll be helpful when showing xref failures. Currently, in error messages' `elements`, we render element name. We could instead/also show original text there so it's easier to reason about.

**Comment by marcoscaceres** [comment]
> Should use data-original-text sometime though. I think it'll be helpful when showing xref failures. 

Yeah, I agree... we can then also show aliased "|" things.

I'll rewrite it.

File: src/core/data-cite.js [hunk 2]
```diff
@@ -181,6 +181,9 @@ export async function run() {
       linkElem(elem, linkProps, citeDetails);
     } else {
       const msg = `Couldn't find a match for "${originalKey}"`;
+      if (elem.dataset?.matchedText) {
+        elem.textContent = elem.dataset?.matchedText;
```
Comment (marcoscaceres) [inline src/core/data-cite.js:185]:
```suggestion
        elem.textContent = elem.dataset.matchedText;
```

File: src/core/data-cite.js [hunk 3]
```diff
@@ -181,6 +181,9 @@ export async function run() {
...
+      }
```
Comment (marcoscaceres) [inline src/core/data-cite.js:186]:
```suggestion
      if (elem.dataset?.matchedText) {
        elem.textContent = elem.dataset.matchedText;
      }
```

**Pull request merged**